google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/apis/GoogleCloudApiManagementPanel.java

```diff
@@ -0,0 +1,278 @@
+  }
+
+  @VisibleForTesting
+  JTable getSelectedApisTable() {
+    return selectedApisTable;
+  }
+
+  @VisibleForTesting
+  JTable getAllApisTable() {
+    return allApisTable;
+  }
+
+  @VisibleForTesting
+  ProjectSelector getProjectSelector() {
+    return projectSelector;
+  }
+
+  private void createUIComponents() {
+    selectedApisTable = new ApisTable();
+    selectedApisTable.setTableHeader(null);
+
+    allApisTable = new ApisTable();
+    allApisTable.setTableHeader(null);
+  }
+
+  private static final class ApisTable extends JBTable {
+    ApisTable() {
+      setDefaultRenderer(CloudLibrary.class, new CloudLibraryRenderer());
+      setDefaultRenderer(Boolean.class, new CloudApiEnablementRenderer());
+    }
+  }
+
+  /** Custom {@link TableModel} for the API enablement table instances. */
+  @VisibleForTesting
+  static final class ApisTableModel implements TableModel {
+
+    private final SortedMap<CloudLibrary, Boolean> librariesToEnabledStatus =
+        new TreeMap<>(Comparator.comparing(CloudLibrary::getName));
+
+    ApisTableModel(Collection<CloudLibrary> libraries) {
+      librariesToEnabledStatus.putAll(Maps.toMap(libraries, this::getEnabledStatus));
+    }
+
+    boolean getEnabledStatus(CloudLibrary library) {
+      // TODO(eshaul) query the IAM API to get the actual status of the API
+      return false;
+    }
+
+    @Override
+    public int getRowCount() {
+      return librariesToEnabledStatus.size();
+    }
+
+    @Override
+    public int getColumnCount() {
+      return 2;
+    }
+
+    @Override
+    public String getColumnName(int columnIndex) {
+      return null;
+    }
+
+    @Override
+    public Class<?> getColumnClass(int columnIndex) {
+      if (columnIndex == 0) {
+        return CloudLibrary.class;
+      }
+
+      return Boolean.class;
+    }
+
+    @Override
+    public boolean isCellEditable(int rowIndex, int columnIndex) {
+      return false;
+    }
+
+    @Override
+    public Object getValueAt(int rowIndex, int columnIndex) {
+      return columnIndex == 0
+          ? librariesToEnabledStatus.keySet().toArray()[rowIndex]
+          : librariesToEnabledStatus.values().toArray()[rowIndex];
+    }
+
+    @Override
+    public void setValueAt(Object aValue, int rowIndex, int columnIndex) {}
+
+    @Override
+    public void addTableModelListener(TableModelListener l) {}
+
+    @Override
+    public void removeTableModelListener(TableModelListener l) {}
+  }
+
+  /**
+   * Custom cell renderer to handle display of the enablement column of the API enablement table. If
+   * the API is not enabled on GCP, then a button is rendered allowing the user to enable the API
+   * from the IDE.
+   */
+  private static final class CloudApiEnablementRenderer extends DefaultTableCellRenderer {
+
+    private final JButton enableButton =
+        new JButton(GctBundle.message("cloud.apis.management.enable.button"));
+
+    CloudApiEnablementRenderer() {
+      super();
+
+      setOpaque(false);
```

Comment:
We probably don't need the first `setOpaque` since the default table renderer is not opaque.